File: 554_Brick_Wall/554_1.scala
import scala.collection.mutable

object Solution {
  def leastBricks(wall: List[List[Int]]): Int = {
    val map = mutable.HashMap[Int, Int](0 -> 0)
    wall.foreach { w =>
      var pos = 0
      for (brick <- w.init) {
        pos += brick
        map(pos) = map.getOrElse(pos, 0) + 1
      }
    }
    if (map.nonEmpty) wall.length - map.values.max else 0
  }
}

File: 554_Brick_Wall/554_1.py
class Solution:
    def leastBricks(self, wall: List[List[int]]) -> int:
        map_ = dict()
        for w in wall:
            pos = 0
            for b in w[:-1]:
                pos += b
                map_[pos] = map_.get(pos, 0) + 1
        return len(wall) - (max(map_.values()) if map_ else 0)
